codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -153,6 +134,10 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
 
     @Override
     public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+        // Initialize the encoder and decoder.
+        encoder.lifecycleManager(this);
```

Comment:
@Scottmitch what exactly is the action you're asking for here?  It seems that this code guarantees they have a consistent lifecycle managers. Or are you suggesting that we ensure the lifecycle managers for both are null in the constructor?